core/src/main/java/nl/nn/adapterframework/http/rest/ApiListenerServlet.java

```diff
@@ -247,6 +249,47 @@ protected void service(HttpServletRequest request, HttpServletResponse response)
 								}
 							}
 						}
+						break;
+					case JWT:
+						String authorizationHeader = request.getHeader("Authorization");
+						if(StringUtils.isNotEmpty(authorizationHeader) && authorizationHeader.contains("Bearer")) {
+							try {
+								Map<String, Object> claimsSet = listener.getJwtValidator().validateJWT(authorizationHeader.substring(7));
+								messageContext.setSecurityHandler(new JwtSecurityHandler(claimsSet, listener.getRoleClaim()));
+								messageContext.put("ClaimsSet", JSONObjectUtils.toJSONString(claimsSet));
+							} catch(Exception e) {
+								log.warn("unable to validate jwt",e);
+								response.sendError(401, e.getMessage());
+								return;
+							}
+						} else {
+							response.sendError(401, "JWT is not provided as bearer token");
+						}
+						String requiredClaims = listener.getRequiredClaims();
+						String exactMatchClaims = listener.getExactMatchClaims();
+						JwtSecurityHandler handler = (JwtSecurityHandler)messageContext.getSecurityHandler();
+						try {
+							handler.validateClaims(requiredClaims, exactMatchClaims);
+							if(StringUtils.isNotEmpty(listener.getRoleClaim())) {
+								List<String> authRoles = listener.getAuthenticationRoleList();
+								if(authRoles != null) {
+									for (String role : authRoles) {
+										if(handler.isUserInRole(role, messageContext)) {
+											userPrincipal = new ApiPrincipal();
+											break;
+										}
+									}
+								} else {
+									userPrincipal = new ApiPrincipal();
+								}
+							} else {
+								userPrincipal = new ApiPrincipal();
+							}
+						} catch(IllegalArgumentException e) {
```

Comment:
I think it's better to use a checked exception in `validateClaims` and not a `IllegalArgumentException`.
Perhaps you could create an `AuthorisationException`?